src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManagerFactory.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    [Shared]
+    [ExportLanguageServiceFactory(typeof(RazorDocumentManager), RazorLanguage.Name, ServiceLayer.Default)]
+    internal class DefaultRazorDocumentManagerFactory : ILanguageServiceFactory
+    {
+        private readonly RazorEditorFactoryService _editorFactoryService;
+        private readonly TextBufferProjectService _projectService;
+
+        [ImportingConstructor]
+        public DefaultRazorDocumentManagerFactory(
+            RazorEditorFactoryService editorFactoryService,
+            TextBufferProjectService projectService)
```

Comment:
This type should also come from Roslyn services. Can be a follow up.

Reply:
Added it to the checklist https://github.com/aspnet/Razor/issues/1799#issue-279962189